stacks-signer/src/signer.rs

```diff
@@ -334,20 +339,6 @@ impl From<SignerConfig> for Signer {
             signer_config.signer_entries.public_keys,
         );
 
-        if let Some(state) = load_encrypted_signer_state(
```

Comment:
Why are we no longer instantiating the signer's state here?  Does this leave open the possibility of creating an invalid `Signer` object whose state is does not reflect the persisted signer state?

Reply:
I've changed this now because the signer needs to know the approved aggregate key to load the right shares. This is not known in the config, and you need a stacks client to fetch it which the signer doesn't have access to.

I appreciate you highlighting this. I am not happy about the current code structure either. I'll add a proper constructor for the Signer so we don't have to be limited to the fields in `SignerConfig` for the initialization logic, and remove the `From<SignerConfig>` implementation which imposes unnecessary restrictions on how signers can be constructed without providing any value anywhere as far as I can see.

Reply:
I've purged this now, in favor of only doing the state reload from the main loop if an aggregate key is not set in the signer - which will always be the case for a new signer.